exec/java-exec/src/main/codegen/templates/ComparisonFunctions.java

```diff
@@ -215,6 +192,36 @@ public void eval() {
     }
   }
 
+  <#-- IS_DISTINCT_FROM function -->
+  @FunctionTemplate(names = {"is_distinct_from", "is distinct from" },
```

Comment:
I am not sure if there is way to differentiate between the function in join condition vs. function in project expr. I don't see any context info in DrillConvertletTable.get() method call. Also the generated code in rewritten case is too much. For following query:
```SELECT INT_col is not distinct from BIGINT_col as col, int_distinct_result FROM cp.`functions/distinct_from.json```

Without rewrite: https://gist.github.com/vkorukanti/e981058f985ed24e6c4ef6b47d670e0f
With rewrite: https://gist.github.com/vkorukanti/d80aa2ba40c65c9215c38ed18b20a685

Sizes may differ after scalar replacement is done, but it is still too much code to simple ```is not distinct from``` function.